integration-cli/check_test.go

```diff
@@ -38,8 +38,12 @@ func (s *DockerRegistrySuite) SetUpTest(c *check.C) {
 }
 
 func (s *DockerRegistrySuite) TearDownTest(c *check.C) {
-	s.reg.Close()
-	s.ds.TearDownTest(c)
+	if s.reg != nil {
```

Comment:
Not sure why this change ? It's not really related to the goal of the PR. Did you have issues with integration-cli tests ?